From 13a9302f319a6c0074d3e22e4fa38e548f73c881 Mon Sep 17 00:00:00 2001
From: Joonwoo Park <joonwpark81@gmail.com>
Date: Tue, 27 Jul 2010 00:53:20 -0400
Subject: [PATCH] IPV4: fib_hash: fix duplicated route issue (patch from
 wl500g.googlecode.com) Backport from 2.6 kernel.org commit
 bd566e7525b5986864e8d6eb5b67640abcd284a9

---
 net/ipv4/fib_hash.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/net/ipv4/fib_hash.c b/net/ipv4/fib_hash.c
index 71a9408677..f5ec4df5a3 100644
--- a/net/ipv4/fib_hash.c
+++ b/net/ipv4/fib_hash.c
@@ -617,6 +617,9 @@ rta->rta_prefsrc ? *(u32*)rta->rta_prefsrc : 0);
 			goto out;
 
 		if (n->nlmsg_flags&NLM_F_REPLACE) {
+			if (fi->fib_treeref > 1)
+				goto out;
+
 			del_fp = fp;
 			fp = &f->fn_next;
 			f = *fp;
-- 
2.11.4.GIT

